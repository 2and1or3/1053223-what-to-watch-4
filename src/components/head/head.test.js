import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {BrowserRouter} from "react-router-dom";

import Head from './head';
import NameSpace from '../../reducer/namespace';

const mockStore = configureStore();

describe(`Render full Head component`, () => {
  it(`Render full Head component without auth`, () => {
    const initialState = {
      [NameSpace.USER]: {
        authStatus: `NO_AUTH`,
      },
    };

    const store = mockStore(initialState);

    const tree = renderer
      .create(
          <Provider store = {store}>
            <BrowserRouter>
              <Head hasUser hasTitle = {`some title`}/>
            </BrowserRouter>
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render full Head component with auth`, () => {
    const initialState = {
      [NameSpace.USER]: {
        authStatus: `AUTH`,
      },
    };

    const store = mockStore(initialState);

    const tree = renderer
      .create(
          <Provider store = {store}>
            <BrowserRouter>
              <Head hasUser hasTitle = {`some title`}/>
            </BrowserRouter>
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

// npm run test.jest -- components/head/head.test
