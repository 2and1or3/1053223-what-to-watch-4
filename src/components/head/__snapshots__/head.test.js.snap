// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render full Head component Render full Head component with auth 1`] = `
<header
  className="page-header user-page__head movie-card__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link "
      href=""
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <h1
    className="page-title user-page__title"
  >
    some title
  </h1>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
    >
      <a
        href="/mylist"
        onClick={[Function]}
      >
        <img
          alt="User avatar"
          height="63"
          src="img/avatar.jpg"
          width="63"
        />
      </a>
    </div>
  </div>
</header>
`;

exports[`Render full Head component Render full Head component without auth 1`] = `
<header
  className="page-header user-page__head movie-card__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link "
      href=""
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <h1
    className="page-title user-page__title"
  >
    some title
  </h1>
  <div
    className="user-block"
  >
    <a
      className="user-block__sign-in"
      href="/login"
      onClick={[Function]}
      style={
        Object {
          "color": "#dfcf77",
          "textDecoration": "none",
        }
      }
    >
      Sign In
    </a>
  </div>
</header>
`;
